src/zepben/evolve/services/common/base_service.py

```diff
@@ -208,12 +243,33 @@ def get_unresolved_reference_mrids(self, bound_resolvers: Union[BoundReferenceRe
         except TypeError:
             resolvers = [bound_resolvers]
 
-        for refs in self._unresolved_references.values():
-            for ref in refs:
-                for resolver in resolvers:
-                    if ref.from_ref is resolver.from_obj and ref.to_mrid not in seen and ref.resolver == resolver.resolver:
-                        seen.add(ref.to_mrid)
-                        yield ref.to_mrid
+        for resolver in resolvers:
+            if resolver.from_obj.mrid not in self._unresolved_references_from:
+                continue
+            for ref in self._unresolved_references_from[resolver.from_obj.mrid]:
+                if ref.to_mrid not in seen and ref.resolver == resolver.resolver:
+                    seen.add(ref.to_mrid)
+                    yield ref.to_mrid
+
+    def get_unresolved_reference_mrids_from(self, mrid: str) -> Generator[str, None, None]:
```

Comment:
So based on our discussion I think get_unresolved_reference_mrids_from and get_unresolved_reference_mrids_to should not return MRIDs but should return the UnresovledReferences themselves. That way you can see what type the unresolved id is. It's probably fine to leave the get_unresolved_reference_mrids_by_resolver as is because you know the type due to the resolver, but I wonder if return the UnresolvedReference instsance is better to just be consistent?